Documentation/benchmarks/etcd-3-watch-memory-benchmark.md

```diff
@@ -0,0 +1,72 @@
+# Watch Memory Usage Benchmark
+
+*NOTE*: The watch features are still being actively developed and we expect the memory usage to change over time. However we do expect that memory usage should not be significantly increased from the figures below.
+
+A primary goal of etcd is to support large numbers of watching. etcd aims to support as many as O(10k) clients, O(100K) watch streams (O(10) streams per client) and O(10M) total watchings (O(100) watching per stream). Thus the goal is to minimize the memory consumption per watching, since it will dominate the overall consumption.
```

Comment:
O(10k) is the big theta notation. We are not ready to give a exact number or a upper limit right now.